src/main/java/com/microsoft/graph/Requests/GraphClientFactory.java

```diff
@@ -0,0 +1,74 @@
+package com.microsoft.graph.Requests;
+
+import com.microsoft.graph.httpcore.CompressionHandler;
+import com.microsoft.graph.httpcore.GraphTelemetryHandler;
+import com.microsoft.kiota.RequestInformation;
+import com.microsoft.kiota.http.KiotaClientFactory;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+
+import javax.annotation.Nonnull;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.Format;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+
+/**
+ * The GraphClientFactory used to create the OkHttpClient.
+ */
+public class GraphClientFactory {
+    private GraphClientFactory() { }
+
+    /**
+     * The default OkHttpClient Builder for Graph.
+     *
+     * @return an OkHttpClient Builder instance.
+     */
+    public static OkHttpClient.Builder create() {
+        return create(new GraphClientOptions());
+    }
+
+    /**
+     * OkHttpClient Builder for Graph with specified Interceptors and optional GraphClientOptions.
+     *
+     * @param interceptors desired interceptors for use in requests.
+     * @param graphClientOptions the GraphClientOptions for use in requests.
+     * @return an OkHttpClient Builder instance.
+     */
+    public static OkHttpClient.Builder create(@Nonnull Interceptor[] interceptors, @Nullable GraphClientOptions graphClientOptions) {
+        OkHttpClient.Builder builder = graphClientOptions == null ? create() : create(graphClientOptions);
+        for(Interceptor interceptor : interceptors) {
+            builder.addInterceptor(interceptor);
+        }
+        return builder;
+    }
+
+    /**
+     * The OkHttpClient Builder with optional GraphClientOptions
+     *
+     * @param graphClientOptions the GraphClientOptions for use in requests.
+     * @return an OkHttpClient Builder instance.
+     */
+    public static OkHttpClient.Builder create(@Nullable GraphClientOptions graphClientOptions) {
```

Comment:
```suggestion
    public static OkHttpClient.Builder create(@Nonnull GraphClientOptions graphClientOptions) {
```
and update to defensive programing below
(if people don't want to provide that, they can use the overload without the parameter)

Reply:
This is nullable because of the use in the Default BaseGraphRequestAdapter which accepts a nullable graphClientOptions parameter 
https://github.com/microsoftgraph/msgraph-sdk-java-core/blob/2a0549ac4bc155fbed4105644c571de3ff579921/src/main/java/com/microsoft/graph/Requests/BaseGraphRequestAdapter.java#L64-L65
Could add a check to see if it's null or not but the call to super() needs to be the very first line of the constructor, thus the only workaround would be to add yet another constructor that takes options and then remove the options from this constructor.

Reply:
right, disregard then.